pkg/apis/nsx.vmware.com/v1alpha1/vpcnetworkconfiguration_types.go

```diff
@@ -19,38 +24,41 @@ const (
 // in a Namespace's VPCNetworkConfiguration, the Namespace will use the value
 // in the default VPCNetworkConfiguration.
 type VPCNetworkConfigurationSpec struct {
-	// PolicyPath of Tier0 or Tier0 VRF gateway.
-	DefaultGatewayPath string `json:"defaultGatewayPath,omitempty"`
-	// Edge cluster path on which the networking elements will be created.
-	EdgeClusterPath string `json:"edgeClusterPath,omitempty"`
 	// NSX-T Project the Namespace associated with.
-	NSXTProject string `json:"nsxtProject,omitempty"`
-	// NSX-T IPv4 Block paths used to allocate external Subnets.
-	// +kubebuilder:validation:MinItems=0
-	// +kubebuilder:validation:MaxItems=5
-	ExternalIPv4Blocks []string `json:"externalIPv4Blocks,omitempty"`
-	// Private IPv4 CIDRs used to allocate Private Subnets.
-	// +kubebuilder:validation:MinItems=0
-	// +kubebuilder:validation:MaxItems=5
-	PrivateIPv4CIDRs []string `json:"privateIPv4CIDRs,omitempty"`
-	// Default size of Subnet based upon estimated workload count.
-	// Defaults to 26.
-	// +kubebuilder:default=26
-	DefaultIPv4SubnetSize int `json:"defaultIPv4SubnetSize,omitempty"`
-	// DefaultSubnetAccessMode defines the access mode of the default SubnetSet for PodVM and VM.
-	// Must be Public or Private.
-	// +kubebuilder:validation:Enum=Public;Private
-	DefaultSubnetAccessMode string `json:"defaultSubnetAccessMode,omitempty"`
+	NsxProject string `json:"nsxProject,omitempty"`
+
+	// VpcConnectivityProfile ID. This profile has configuration related to create VPC transit gateway attachment.
+	VpcConnectivityProfile string `json:"vpcConnectivityProfile,omitempty"`
+
+	// The path of the configuration profile of the VPC services. This will be an collection of default dhcp and subnet profiles. The default vpc service profile will be created as part of new project create workflow. That will be used as the default for all VPC created under that project. The default value will be project specific default VPC profile.
+	VpcServiceProfile string `json:"vpcServiceProfile,omitempty"`
+
+	// Private IPs.
+	PrivateIPs []string `json:"privateIPs,omitempty"`
+
 	// ShortID specifies Identifier to use when displaying VPC context in logs.
-	// Less than or equal to 8 characters.
+	// Less than equal to 8 characters.
 	// +kubebuilder:validation:MaxLength=8
 	// +optional
 	ShortID string `json:"shortID,omitempty"`
+
```

Comment:
Could we group all VPC config parameters: Project, Profiles, PrivateIPs, ShortID, LBSize to a struct like VPCConfig to be consistent with VAPI?

Reply:
@lxiaopei @dantingl 
If it requires considerable changes, then I am fine to keep the current way. Anyway, this is just an internal CRD, not for end users/applications.

Reply:
Do you think maybe in future there could be other configurations added into this CRD?

Reply:
Yes we can. But still no issue to group parameters for auto VPC creation to one struct?

Reply:
I submitted an issue #673 and I think we can implement it with another PR?